### DISCUSSION TITLE
How do you compile standalone excalidraw with webpack ?

### DISCUSSION THREAD

**Discussion by gitubbi** [excalidraw/excalidraw#6617] [Q&A] [answered]
Hello,

Could please tell me what is the command to compile the excalidraw sources with webpack to produce the file excalidraw.production.min.js.

I have downloaded the exalidraw-main folder on my pc and I can build the source without problem with "npm run build", but this versions is to be used with a nodejs server as far as i understand.

In my case I need the standalone version with everything included in the file "excalidraw.production.min.js" you found in the dist folder of the release. I have tried different commands taken from differents webpack tutos like "npx webpack", but it always makes error. 

I wish to be able to use on my local server the new colorpicker added recently and can't wait the next release 0.15.2 :).

Thank you for any help.

**Comment by crapthings** [comment]
same here, how?

**Comment by johnrbnsn** [comment]
I'm looking for the same thing :)

**Comment by yalogica** [comment, marked as answer]
Well. It depends on the source code package. At the moment I managed to build version [v0.16.0](https://github.com/excalidraw/excalidraw/releases/tag/v0.16.0)
1) Download the package and unzip it to the folder "excalidraw-0.16.0"
2) Run shell, go to the folder "excalidraw-0.16.0"
3) Run "yarn install", then "yarn run build"
4) Go to the folder "excalidraw-0.16.0/src/packages/excalidraw/"
5) Run "yarn install", then "yarn run build:umd"
6) The dist folder should appear, it should contain "excalidraw.development.js" and "excalidraw.production.min.js" and etc

**Reply by johnrbnsn** [reply]
@yalogica , thanks! I was able to build using your method. Do you know how I can use this in a vite project after I've got it built? I tried putting it inside a `components` directory and importing into my project like this `import { Excalidraw } from './components/excalidraw/excalidraw.development.js'`, but got an error saying Excalidraw was not exported.

**Reply by yalogica** [reply]
@johnrbnsn. Well, let me show you my setup
1) I have this structure 
    d:\excalidraw-embed\local_modules\excalidraw-0.17.0\src\packages\excalidraw\
    d:\excalidraw-embed\node_modules\\*
    d:\excalidraw-embed\public
    d:\excalidraw-embed\src
    d:\excalidraw-embed\package.json
2) To use local library from local_modules I installed yalc package - https://github.com/wclr/yalc
    **npm i yalc -g**
3) Then in "d:\excalidraw-embed\local_modules\excalidraw-0.17.0\src\packages\excalidraw\" I run **"yalc publish"**
4) In "d:\excalidraw-embed\" I run **"yalc add @excalidraw/excalidraw --link"**. It will add a local package to package.json
5) Now I can use **import { Excalidraw } from "@excalidraw/excalidraw";** in my src files.
5) If I change the excalidraw source I need to run **"yarn run build:umd"** and **"yalc push"** inside "d:\excalidraw-embed\local_modules\excalidraw-0.17.0\src\packages\excalidraw\"
